agent/api/task/task.go

```diff
@@ -459,66 +460,37 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 }
 
 // initNetworkMode initializes/infers the network mode for the task and assigns the result to this task's NetworkMode field.
-// This is necessary because ACS doesn't send the actual network mode declared in task def.
-// TODO [SC]: Instead of doing this, it's perhaps better/easier to just ask ACS to send network mode from task definition. Using this in the meantime to avoid being blocked.
-func (task *Task) initNetworkMode() {
-	if len(task.ENIs) > 0 {
+// ACS is streaming down this value with task payload. In case of docker bridge mode task, this value might be left empty
+// as it's the default task network mode.
+func (task *Task) initNetworkMode(acsTaskNetworkMode *string) {
+	switch aws.StringValue(acsTaskNetworkMode) {
+	case AWSVPCNetworkMode:
 		task.NetworkMode = AWSVPCNetworkMode
-		return
-	}
-	for _, c := range task.Containers {
-		// ACS sets task def network mode to each and every customer container docker host config, so the presence
-		// of one container with a given network mode is enough to infer the task network mode
-		switch c.GetNetworkModeFromHostConfig() {
-		case BridgeNetworkMode:
-			task.NetworkMode = BridgeNetworkMode
-			return
-		case HostNetworkMode:
-			task.NetworkMode = HostNetworkMode
-			return
-		case "":
-			// the absence of a network mode means the default docker network mode is used, which, for ECS it's bridge.
-			// for future proofing, let's keep inspecting the rest of the containers for this particular case.
-			continue
-		}
+	case HostNetworkMode:
+		task.NetworkMode = HostNetworkMode
+	case BridgeNetworkMode, "":
+		task.NetworkMode = BridgeNetworkMode
+	default:
```

Comment:
In case of `default`, dont we need to set this to `Bridge`?

Reply:
Wouldnt `validateAttachment ` again fail when the network mode is empty?

Reply:
We won't expect to get an SC attachment should the network mode is `None` - but I will add a case for `None` network mode as well.